packages/admin/admin/src/form/FinalFormSelect.tsx

```diff
@@ -45,28 +46,59 @@ export const FinalFormSelect = <T,>({
 
     if (children) {
         return (
-            <Select {...rest} endAdornment={selectEndAdornment} name={name} onChange={onChange} value={value} onFocus={onFocus} onBlur={onBlur}>
+            <Select
+                {...rest}
+                endAdornment={selectEndAdornment}
+                name={name}
+                onChange={onChange}
+                value={value}
+                onFocus={onFocus}
+                onBlur={onBlur}
+                multiple={multiple}
+            >
                 {children}
             </Select>
         );
     }
 
-    if (value && options) {
-        value = options.reduce((previousOption, option) => (getOptionSelected(option, value) ? option : previousOption), value);
-    }
-
     return (
-        <Select {...rest} endAdornment={selectEndAdornment} name={name} onChange={onChange} value={value} onFocus={onFocus} onBlur={onBlur}>
-            {options.length === 0 && (loading || value) && (
-                <MenuItem value={value as any} key={JSON.stringify(value)}>
-                    {loading ? <CircularProgress size={20} /> : getOptionLabel(value)}
-                </MenuItem>
-            )}
-            {options.map((option: T) => (
-                <MenuItem value={option as any} key={JSON.stringify(option)}>
-                    {getOptionLabel(option)}
-                </MenuItem>
-            ))}
-        </Select>
+        <div>
+            <Select
+                {...rest}
+                endAdornment={selectEndAdornment}
+                name={name}
+                onChange={(event) => {
+                    const value = event.target.value;
+                    onChange(
+                        Array.isArray(value)
+                            ? value.map((v) => options.find((i) => getOptionValue(i) == v))
+                            : options.find((i) => getOptionValue(i) == value),
+                    );
+                }}
+                value={Array.isArray(value) ? value.map((i) => getOptionValue(i)) : getOptionValue(value)}
+                onFocus={onFocus}
+                onBlur={onBlur}
+                multiple={multiple}
+            >
+                {options.length === 0 &&
+                    (loading || value) &&
+                    (Array.isArray(value) ? (
+                        value.map((v) => (
+                            <MenuItem value={getOptionValue(v)} key={getOptionValue(v)}>
+                                {loading ? <CircularProgress size={20} /> : getOptionLabel(v)}
```

Comment:
This would render multiple circular progresses if the select is loading but has a value... is this what we want?

However, is there ever a scenario in which loading might we true and we also have a value?

Reply:
you are right, that was showing multiple (one per selected value) loading animations.

This however changes behaviour also of non-multiple select, see here:

https://github.com/vivid-planet/comet/assets/50764/41d3b423-6c19-4c35-ace3-6b125d2e1613

previous behaviour:

https://github.com/vivid-planet/comet/assets/50764/8a4beef2-4a5a-44e7-b3f9-79c45d87e8de


any alternative ideas?

Reply:
hm, seems not so easy to do https://github.com/mui/material-ui/issues/17799 without custom styling

Reply:
I added this commit, now it looks like the async autocomplete: [Make async-select loading-state look like in async-autocomplete](https://github.com/vivid-planet/comet/pull/1099/commits/7b885c7e266a93ccc4084fea1612c029ba72247d)

https://github.com/vivid-planet/comet/assets/6264317/e30af084-8d49-45bc-895b-c2ea10ef75ca